src/CADL.Extension/Emitter.Csharp/src/lib/model.ts

```diff
@@ -215,7 +216,8 @@ export function getInputType(
     program: Program,
     type: Type,
     models: Map<string, InputModelType>,
-    enums: Map<string, InputEnumType>
+    enums: Map<string, InputEnumType>,
+    name: string = ""
```

Comment:
This is a potential error prone:
- We call `getInputType` from other `getXxxType` methods, but in these methods name will always be default.
- Two parameters may have the same name but different set of stings in the enum.